### DISCUSSION TITLE
Disable autoexpansion of snippets when completing

### DISCUSSION THREAD

**Discussion by asmodeus812** [neoclide/coc.nvim#4955] [Q&A]
According to the docs we could do that with

```txt
	- "disableSnippetCompletion": Disable snippet completion feature for
	  this languageserver.
```

However having this in my coc settings, does not work

```json
    "languageserver.disableSnippetCompletion": true,           // not working / applied for all servers 
    "languageserver.java.disableSnippetCompletion": true,      // not working even with specific server
```

What is the correct way to disable the expansion on functions/symbols arguments/paramters, but retain / keep the expansion of snippets.

I would like to avoid the null arguments (see below) being automatically added by coc
```java
   Collections.addAll(null, null)
```

**Comment by fannheyward** [comment]
The `disableSnippetCompletion` configuration is used for `languageserver` section, something like 

```json
"languageserver": {
  "clangd": {
    "command": "clangd",
    "disableSnippetCompletion": true,
    "rootPatterns": ["compile_flags.txt", "compile_commands.json"],
    "filetypes": ["c", "cc", "cpp", "c++", "objc", "objcpp"]
  }
}
```

**Reply by fannheyward** [reply]
For the `Collections.addAll` issue, are you using coc-java? I don't think coc-java supports to disable snippets, the `disableSnippetCompletion` won't work for it.

**Reply by asmodeus812** [reply]
Is it not possible to simply disable the client capability for snippets? Can't cocs client simply announce it does not support snippet expansion ? At least with the nvim native lsp, that is how we used to disable certain capabilities globally.

**Reply by fannheyward** [reply]
Use `suggest.snippetsSupport` to disable snippets completion.

**Reply by asmodeus812** [reply]
@fannheyward well yes but this breaks code snippets provided by the lang server, `for (${1:int} ${2:i} = ${3:0}; ${2:i} < ${4:array.length}; ${2:i}++)`. Does coc support integration with something like luasnip and friendly snippets ?

Edit: I did try coc-snippets, but since snippet completion is disabled, it would not expand code snippets like it does not for ones coming from the language server itself.

If there is a way to tell coc to expand the snippets (on tab/enter) but not the function params / arguments that would be great

**Reply by fannheyward** [reply]
coc-snippets can use friendly snippets.

> tell coc to expand the snippets (on tab/enter) but not the function params / arguments

There're both snippets. Needs a configuration to disable server returns params as snippets, `disableSnippetCompletion` is used for this.
